iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -435,6 +365,34 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
         newOffset.y = max(newOffset.y, 0)
         return newOffset
     }
+
+    // MARK: - Scroll/Animation Helper Functions
+
+    /// Gets the focus position of the item associated with the passed in date.
+    /// If there is no item with that date, it finds the next oldest item.
+    /// If there is still no item found, then nil is returned.
+    private func getFocusPositionOfItem(with date: Date) -> CGFloat? {
```

Comment:
This function looks complicated, but the idea is pretty simple. Get an item that matches the passed in date.